com.unity.shadergraph/Editor/GraphDeltaRegistry/FunctionDefinitions/FieldType.cs

```diff
@@ -0,0 +1,98 @@
+namespace com.unity.shadergraph.defs {
+
+    // ----------
+    // Type
+
+    enum Primitive { Bool, Float, Int, Dynamic, }
+
+    enum Precision { Fixed, Half, Full, Dynamic, }
+
+    enum Width { One, Two, Three, Four, Any, }
+
+    enum Height { One, Two, Three, Four, Any, }
+
+    public readonly struct Type
```

Comment:
I know this is in our namespace, but Type for the name here is probably too generic since this only covers the built-in matrix/vector/scalar types available in shading languages.